pkg/network/tracer/offsetguess/tracer.go

```diff
@@ -497,21 +605,42 @@ func (t *tracerOffsetGuesser) checkAndUpdateCurrentOffset(mp *ebpf.Map, expected
 			break
 		}
 		t.status.Offset_socket_sk++
+		// no overlaps because only field offset from `struct socket`
 	case GuessSKBuffSock:
+		t.status.Offset_sk_buff_sock, overlapped = skipOverlaps(t.status.Offset_sk_buff_sock, t.skBuffRanges())
+		if overlapped {
+			// adjusted offset from eBPF overlapped with another field, we need to check new offset
+			break
+		}
+
 		if t.status.Sport_via_sk_via_sk_buf == htons(expected.sportFl4) && t.status.Dport_via_sk_via_sk_buf == htons(expected.dportFl4) {
 			t.logAndAdvance(t.status.Offset_sk_buff_sock, GuessSKBuffTransportHeader)
 			break
 		}
 		t.status.Offset_sk_buff_sock++
+		t.status.Offset_sk_buff_sock, _ = skipOverlaps(t.status.Offset_sk_buff_sock, t.skBuffRanges())
```

Comment:
Do we need the check after increment? Wouldn't the check at the beginning of a case get hit next time around anyway?

Reply:
You've then wasted an eBPF cycle on an offset you don't want.